actors/builtin/miner/miner_state.go

```diff
@@ -133,28 +133,29 @@ type WorkerKeyChange struct {
 	EffectiveAt abi.ChainEpoch
 }
 
+// possible states
+// (DealIDs > 0, UnsealedCID != nil) => CommD provided by SP, needs to be checked
+// (DealIDs > 0, UnsealedCID == nil) => Old method or migration, DealIDs checked at PreCommit (Depracated)
+// (DealIDs == 0, UnsealedCID == nil) => NoDeals, CommD of empty sector
+// (DealIDs == 0, UnsealedCID != 0) future approach, fail for now
 // Information provided by a miner when pre-committing a sector.
 type SectorPreCommitInfo struct {
-	SealProof       abi.RegisteredSealProof
-	SectorNumber    abi.SectorNumber
-	SealedCID       cid.Cid `checked:"true"` // CommR
-	SealRandEpoch   abi.ChainEpoch
-	DealIDs         []abi.DealID
-	Expiration      abi.ChainEpoch
-	ReplaceCapacity bool // Whether to replace a "committed capacity" no-deal sector (requires non-empty DealIDs)
-	// The committed capacity sector to replace, and it's deadline/partition location
-	ReplaceSectorDeadline  uint64
-	ReplaceSectorPartition uint64
-	ReplaceSectorNumber    abi.SectorNumber
+	SealProof     abi.RegisteredSealProof
+	SectorNumber  abi.SectorNumber
+	SealedCID     cid.Cid `checked:"true"` // CommR
+	SealRandEpoch abi.ChainEpoch
+	DealIDs       []abi.DealID
+	Expiration    abi.ChainEpoch
+	UnsealedCID   *cid.Cid `checked:"true"`
 }
 
 // Information stored on-chain for a pre-committed sector.
 type SectorPreCommitOnChainInfo struct {
-	Info               SectorPreCommitInfo
-	PreCommitDeposit   abi.TokenAmount
-	PreCommitEpoch     abi.ChainEpoch
-	DealWeight         abi.DealWeight // Integral of active deals over sector lifetime
-	VerifiedDealWeight abi.DealWeight // Integral of active verified deals over sector lifetime
+	Info             SectorPreCommitInfo
+	PreCommitDeposit abi.TokenAmount
+	PreCommitEpoch   abi.ChainEpoch
+	//DealWeight         abi.DealWeight // Integral of active deals over sector lifetime
+	//VerifiedDealWeight abi.DealWeight // Integral of active verified deals over sector lifetime
```

Comment:
If we do this, we'll need to migrate existing pre-commits in state. In the FIP I'd suggested just setting to zero.

The migration might be reasonable since there won't be many, and the clarity of not having this field is good. So 👍 for now.

Reply:
We need to migrate the state either way to add the CommD.